mr_project/mr_app/views.py

```diff
@@ -1,9 +1,231 @@
+import json
+
 from django.shortcuts import render
 
 # Create your views here.
+from time import time
+from django.http import HttpResponse, HttpResponseBadRequest, HttpResponseNotFound, JsonResponse
+import neo4j
+from django.views.decorators.csrf import csrf_exempt
+from requests import codes
+from . import DRIVER, EMOTIONS, USERS
+from datetime import datetime
+
+#WRITE TRANSACTION FUNCTIONS
+def create_user_tx(tx, name):
+    result = tx.run('CREATE (u:User{name:$name}) RETURN u.name AS username', name=name)
+    record = result.single()
+    result.consume()
+    return record['username']
+
+
+def create_emotion_tx(tx, name):
+    result = tx.run('CREATE (em:Emotion {name:$name})  RETURN em.name AS emotion_label', name=name)
+    record = result.single()
+    result.consume()
+    return record['emotion_label']
+
+
+def create_epoch_relationship_between_user_and_emotion(tx, emotion, user, epoch_timestamp):
+    result = tx.run(f'MATCH (em:Emotion {{name:"{emotion}"}}), (u:User {{name:"{user}"}}) CREATE (u)-[ts:Timestamp{{epoch:"{epoch_timestamp}"}}]->(em) RETURN ts.epoch as seconds')
+    record = result.single()
+    result.consume()
+    return record['seconds']
+
+
+#EFFECTIVE READ TRANSACTIONS TO CHECK IF NODES EXIST OR NOT
+def check_for_user(driver_arg, user_name):
+    with driver_arg.session() as session:
+        result = session.run("MATCH (n:User{ name:$user_name }) RETURN n.name AS name", user_name=user_name)
+        values = result.values()
+        if not values:
+            return False
+        else:
+            return True
+
+
+def check_for_emotion(driver_arg, emotion_name):
+    with driver_arg.session() as session:
+        result = session.run("MATCH (n:Emotion{ name:$emotion_name }) RETURN n.name AS name", emotion_name=emotion_name)
+        values = result.values()
+        if not values:
+            return False
+        else:
+            return True
+
+
+def fetch_epochs_of_user_emotion_pair(driver_arg, user_name, emotion_name):
+
+    with driver_arg.session() as session:
+        result = session.run('''MATCH (u)-[ts]->(em) WHERE u.name = $user_name AND em.name = $emotion_name RETURN 
+                             ts.epoch AS timestamps''', user_name=user_name, emotion_name=emotion_name)
+
+        timestamp_list = []
+        for record in result:
+            timestamp_list.append(record["timestamps"])
+
+    one_emotion_dict = {
+        "emotion": emotion_name,
+        "timestampList": timestamp_list
+    }
+    return one_emotion_dict
+
+
+def fetch_all_emotions_of_one_user(driver_arg, user_name):
+    with driver_arg.session() as session:
+        result = session.run('''MATCH (u)-->(em) WHERE u.name = $user_name  RETURN 
+                             em.name AS emotions''', user_name=user_name)
+        emotion_list = []
+        for record in result:
+            if not record["emotions"] in emotion_list:
+                emotion_list.append(record["emotions"])
+    return emotion_list
```

Comment:
This can be simplified. `Set`s only contain distinct values so converting the `List` to a `Set` will remove duplicates. 
```python
return Set([record['emotions'] for record in result])
```

Reply:
MORE LISTS!!?!?!?!?!?!

Reply:
do tests on if sets mess up the lists

Reply:
this code doesn't work either